src/json-editor.component.ts

```diff
@@ -65,33 +67,27 @@ import { JsonEditorConfig, Preview, SchemaValidationErrors, JsonPatch } from './
   changeDetection: ChangeDetectionStrategy.OnPush
 })
 
-export class JsonEditorComponent extends AbstractTrackerComponent implements OnInit {
+export class JsonEditorComponent extends AbstractTrackerComponent implements OnChanges, OnInit {
 
   @Input() config: JsonEditorConfig;
   @Input() record: Object;
   @Input() schema: any;
-  @Input() set errorMap(errors: SchemaValidationErrors) {
-    this._errorMap = errors;
-    this.errorsService.externalErrors = this.errorMap;
-  }
+  @Input() errorMap: SchemaValidationErrors;
   @Input() jsonPatches: Array<JsonPatch>;
-
-  get errorMap(): SchemaValidationErrors {
-    return this._errorMap;
-  }
-
   @Input() templates: { [templateName: string]: TemplateRef<any> } = {};
 
-  @Output() onRecordChange: EventEmitter<Object> = new EventEmitter<Object>();
+  @Output() recordChange = new EventEmitter<Object>();
 
-  private _errorMap: SchemaValidationErrors = {};
   _record: Map<string, any>;
   tabNames: Array<string>;
   previews: Array<Preview>;
   isPreviewerHidden: boolean;
   isBottomConsoleOpen = false;
   bottomConsoleActiveTab = '';
 
+  // used to decide if the [record] is change caused by recordChange.emit or parent component
+  private lastEmittedRecord: Object;
```

Comment:
This is need because, ngOnChanges is triggered also when `recordChange.emit` is called.